CompetitionEvaluation.py

```diff
@@ -56,47 +56,44 @@ def structure_data(observed: pd.DataFrame, predictions: pd.DataFrame, draw_colum
 
     # The samples must be named "member" and the outcome variable needs to be named the same in xs.crps_ensemble()
     
-    predictions = predictions.reset_index()
-    observed = observed.reset_index()
+    if predictions.index.names != [None]:
+        predictions = predictions.reset_index()
+    if observed.index.names != [None]:
+        observed = observed.reset_index()
 
-    if "index" in observed.columns:
-        observed = observed.drop(columns = ["index"])
-    if "index" in predictions.columns:
-        predictions = predictions.drop(columns = ["index"])
+    # assert "month_id" in observed.columns,  f"'month_id' column not found in observed data. Columns in data: {observed.columns}."

```

Comment:
It will be nice to remove the commented code and only keep documentation as comments in the functions for better readability and focus.